src/server/scripts/Spells/spell_quest.cpp

```diff
@@ -1974,25 +1974,23 @@ class spell_q13264_thats_abominable : public SpellScriptLoader
             {
                 PreventHitDefaultEffect(effIndex);
 
-                if (Unit* unit = GetHitUnit())
+                if (Creature* creature = GetHitCreature())
                 {
-                    ObjectGuid objGuid = unit->GetGUID();
-
                     // Except Player
-                    if (objGuid.IsPlayer())
+                    if (creature->GetGUID().IsPlayer())
                         return;
 
                     if (Unit* charmer = GetCaster()->GetCharmerOrOwner())
                         if (Player* player = charmer->ToPlayer())
                             if (player->GetQuestStatus(QUEST_THATS_ABOMINABLE) == QUEST_STATUS_INCOMPLETE)
-                                if (GiveCreditIfValid(player, &objGuid))
-                                    unit->KillSelf();
+                                if (GiveCreditIfValid(player, creature))
+                                    creature->KillSelf();
                 }
             }
 
-            bool GiveCreditIfValid(Player* player, ObjectGuid* obj)
+            bool GiveCreditIfValid(Player* player, Creature* creature)
             {
-                uint32 entry = obj->GetEntry();
+                uint32 entry = creature->GetGUID().GetEntry();
```

Comment:
Creature inherites from Object and Object has a GetEntry() method that returns the same info you need (the NPC ID in this case), so calling creature->GetEntry() is enough.